panoptes_client/panoptes.py

```diff
@@ -231,9 +233,16 @@ def get_csrf_token(self):
 
     def get_bearer_token(self):
         if not self.bearer_token or self.bearer_expires > datetime.now():
-            if not self.logged_in:
+
+            if self.client_secret:
+                grant_type = 'client_credentials'
+            elif self.password:
+                grant_type = 'password'
+
+            if not (self.logged_in and grant_type == 'client_credentials'):
                 if not self.login():
                     return
```

Comment:
You probably don't want to call `self.login()` at all if you're using the client secret for authentication?